conf/modules.config

```diff
@@ -1,337 +1,838 @@
+    GENERAL
+========================================================================================
+*/
 
-        /*
-        ========================================================================================
-            ALIGNMENT
-        ========================================================================================
-        */
+process {
+    publishDir = [
+        path: { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" },
+        mode: 'copy',
+        saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+    ]
+
+    withName: CUSTOM_DUMPSOFTWAREVERSIONS {
+        publishDir = [
+            path: { "${params.outdir}/pipeline_info" },
+            mode: 'copy',
+            pattern: '*_versions.yml'
+        ]
+    }
+}
 
-        // ALIGN BT2 - SPIKE-IN
-        "bowtie2_spikein_align" {
-            args          = "--end-to-end --very-sensitive --no-overlap --no-dovetail --no-mixed --no-discordant --phred33 -I 10 -X 700"
-            suffix        = ".spikein"
-            publish_files = false
-        }
-        "samtools_spikein_sort" {
-            suffix        = ".spikein.sorted"
-            publish_files = false
-        }
+/*
+========================================================================================
+    INPUT
+========================================================================================
+*/
 
-        // ALIGN BT2
-        "bowtie2_align" {
-            args          = "--end-to-end --very-sensitive --no-mixed --no-discordant --phred33 -I 10 -X 700"
-            suffix        = ".target"
-            publish_files = false
-        }
-        "samtools_sort" {
-            suffix        = ".target.sorted"
-            publish_files = false
+if(params.run_input_check) {
+    process {
+        withName: 'NFCORE_CUTANDRUN:CUTANDRUN:INPUT_CHECK:SAMPLESHEET_CHECK' {
+            publishDir = [
+                path: { "${params.outdir}/pipeline_info" },
+                mode: 'copy',
+                saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+            ]
         }
+    }
+}
 
-        /*
-        ========================================================================================
-            Q-FILTERTING
-        ========================================================================================
-        */
+/*
+========================================================================================
+    GENOME PREPARATION
+========================================================================================
+*/
 
-        "samtools_view_qfilter" {
-            suffix        = ".target.filtered"
-            publish_files = false
+if (params.run_genome_prep) {
+    process {
+        withName: '.*PREPARE_GENOME:GUNZIP_.*' {
+            publishDir = [
+                path: { "${params.outdir}/00_genome" },
+                mode: 'copy',
```

Comment:
why not parameterize the mode to `mode: params.publish_dir_mode` ?

Reply:
good point!

Reply:
updated, great spot